synapse/handlers/appservice.py

```diff
@@ -281,7 +281,13 @@ async def _handle_presence(
         presence_source = self.event_sources.sources["presence"]
         from_key = await self.store.get_type_stream_id_for_appservice(
             service, "presence"
-        )
+        )  # type: Optional[int]
+
+        if from_key == 0:
```

Comment:
A test for an appservice `stream_id` of 0 would be good to have ;)

Reply:
We currently have 0 tests for the handler class and as we talked about last time, writing tests is a significant amount of work for it. We could wait until I have the time, but atm this is broken for most people which is not great.

Reply:
I have realised that this code is actually superflous. We are only fetching stream_id 0 for a single user, so the performance impact is basically zero anyway. To keep this code clean, I'm going to remove it and only include the LIMIT code.